### PR TITLE
DOCS-6929 - adding step functions enhanced metrics page

### PR DESCRIPTION
**Pull request by cswatt**
### What does this PR do? What is the motivation?
Adds a page for enhanced step functions metrics

### Merge instructions

- [ ] Please merge after reviewing

### CODE REVIEW

File: content/en/serverless/step_functions/enhanced-metrics.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+: Counts the total number of tasks that have failed.
+
+`aws.states.enhanced.task.execution.task_duration`
+: Distribution of the durations of individual tasks.
+
+`aws.states.enhanced.task.execution.tasks_timed_out`
+: Counts the total number of tasks that timed out.
+
+`aws.states.enhanced.state.run_duration`
+: Gauge for durations of a state’s runs.
```
Comment (hestonhoffman) [inline content/en/serverless/step_functions/enhanced-metrics.md:38]:
```suggestion
: Gauge for durations of a state's runs.
```

File: content/en/serverless/step_functions/enhanced-metrics.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+`aws.states.enhanced.state.duration`
+: Gauge for duration of a state’s runs, including retries.
```
Comment (hestonhoffman) [inline content/en/serverless/step_functions/enhanced-metrics.md:41]:
```suggestion
: Gauge for duration of a state's runs, including retries.
```

**Review by hestonhoffman — APPROVED** [review]
Edits to satisfy Vale

File: content/en/serverless/step_functions/enhanced-metrics.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+`aws.states.enhanced.state.duration`
+: Gauge for duration of a state's runs, including retries.
+
+`aws.states.enhanced.state.invocation_count`
+: Count of the number of times a state is invoked.
+
+`aws.states.enhanced.state.retry_count`
+: Gauge of the number of retries for a state.
+
+[1]: /serverless/aws_lambda/metrics#enhanced-lambda-metr
```
Comment (DarcyRaynerDD) [inline content/en/serverless/step_functions/enhanced-metrics.md:49]:
```suggestion
[1]: /serverless/aws_lambda/metrics#enhanced-lambda-metrics
```

Comment (cswatt) [reply content/en/serverless/step_functions/enhanced-metrics.md:49]: ahhhh i'm going to open a new PR to make this fix

**Pull request merged**